gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -170,6 +195,19 @@ private void scheduleSpecsFromCatalog() {
     }
   }
 
+  /**
+   * In DR-mode, the running {@link FlowSpec} will all be cancelled and rescheduled.
+   * We will need to make sure that running {@link FlowSpec}s' state are cleared, and corresponding running jobs are
+   * killed before rescheduling them.
+   * @param drUris The uris that applicable for DR discovered from FlowCatalog.
+   */
+  private void clearRunningFlowState(Iterator<URI> drUris) {
```

Comment:
Should the DR related failover handling be moved to a separate class? For now, you are triggering it based on config. In the future, we might have a REST end-point or some other mechanism for triggering a fail over, without needing a service restart.

Reply:
Actually the structure is pretty isolated since all additional thing added is calling this method. It shouldn't be very hard to migrate to a RESTful design.